core/org.openjdk.jmc.flightrecorder/src/main/java/org/openjdk/jmc/flightrecorder/stacktrace/tree/Node.java

```diff
@@ -77,25 +73,13 @@ public static Node newRootNode(AggregatableFrame rootFrame) {
 	}
 
 	public Node(Node parent, AggregatableFrame frame) {
-		this.nodeId = computeNodeId(parent, frame);
 		this.parent = parent;
 		this.frame = frame;
 		if (frame == null) {
 			throw new NullPointerException("Frame cannot be null!");
 		}
 	}
 
-	private static Integer computeNodeId(Node parent, AggregatableFrame frame) {
-		return Objects.hash(parent != null ? parent.getNodeId() : null, frame.hashCode());
-	}
-
-	/**
-	 * @return the unique identifier associated with this node.
-	 */
-	public Integer getNodeId() {
-		return nodeId;
-	}
```

Comment:
**issue:** This is a breaking change. And should be reverted.

Reply:
Oh, did someone add a call to this? That's unfortunate. Overall this was not that important for performance. It was however also used in the equals method, which I think was wrong. The nodeId is computed using hashing so in theory there might be collisions. Using that for equals would mean Nodes are only compared to be "kind of" equal and not exactly equal.  
I did however also never find a reference where the equals would have been called. I even had a breakpoint in it that never was hit.

Reply:
It's probable. This could be used in tests, or else. The core libraries are distributed on maven, so whether it is used or not such API changes should be avoided.

Reply:
Oh that's what you meant. I wasn't aware that all of jmc is also published to maven. Then it was obviously wrong to just delete part of the public API.

Reply:
> I wasn't aware that all of jmc is also published to maven.

Only the core libraries.

Reply:
Hi, what's the status here? Should we file an issue to revert getNodeId()? Or is it considered alright?

Reply:
I probably am not going to have time doing anything on this for some time.  
I also I currently think that, while technically one should always stay backwards compatible just to ease migrations and preserve the willingness of users to upgrade, it should be ok in this case. Given the major version jump from 8.x currently published to maven und the development version of 9.0.0 in this repo, which would in theory allow for incompatible changes.

Reply:
> I probably am not going to have time doing anything on this for some time. I also I currently think that, while technically one should always stay backwards compatible just to ease migrations and preserve the willingness of users to upgrade, it should be ok in this case. Given the major version jump from 8.x currently published to maven und the development version of 9.0.0 in this repo, which would in theory allow for incompatible changes.

I agree. Leave it be unless somebody complains with more thorough reasoning. If we were to bring back getNodeId(), it should be deprecated anyway.